### PR TITLE
Improvement: (txstream) extend chopper to support bigger messages

### PR DESCRIPTION
**Pull request by jorgemmsilva**
# Description of change

Please write a summary of your changes and why you made them. Be sure to reference any related issues by adding `fixes # (issue)`.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation Fix

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/txstream/chopper/chopper_test.go [hunk 1]
```diff
@@ -115,10 +115,10 @@ func TestMaxTangleMessage(t *testing.T) {
 
 //goland:noinspection ALL
 func TestTooLongData(t *testing.T) {
-	tooLong := 1000000
+	tooLong := int(1.5 * 1024 * 1024) // 1.5 MB
 	short := 10000
 	chunk := 500
-	assert.True(t, tooLong/chunk+1 > 255)
+	assert.True(t, tooLong/chunk >= 255)
```
Comment (dessaya) [inline packages/txstream/chopper/chopper_test.go:121]: this 255 is `maxNChunks`?

**Review by dessaya — APPROVED** [review]

File: packages/txstream/chopper/chopper.go [hunk 2]
```diff
@@ -13,10 +13,11 @@ import (
 
 const (
 	// for the final data packet to be not bigger than tangle.MaxMessageSize
-	// 4 - chunk id, 1 seq nr, 1 num chunks, 2 - data len
-	chunkHeaderSize     = 4 + 1 + 1 + 2
+	// 4 - chunk id, 1 seq nr, 2 num chunks, 2 - data len
+	chunkHeaderSize     = 4 + 1 + 2 + 2
```
Comment (kape1395) [inline packages/txstream/chopper/chopper.go:17]: Seq Nr must be 2 bytes as well, if I understand it correctly.

File: packages/txstream/chopper/chopper.go [hunk 3]
```diff
@@ -13,10 +13,11 @@ import (
...
 	maxTTL              = 5 * time.Minute
 	cleanupLoopInterval = 10 * time.Second
+	maxNChunks          = 2200
```
Comment (kape1395) [inline packages/txstream/chopper/chopper.go:20]: Why not 2^16?

**Review by kape1395 — CHANGES_REQUESTED** [review]

File: packages/txstream/chopper/chopper.go (hunk 3 above)
Comment (jorgemmsilva) [reply packages/txstream/chopper/chopper.go:20]: I was aiming for a max size of ~1MB. But we can indeed use all the capacity. updating it 👍

**Review by capossele — APPROVED** [review]

**Review by kape1395 — APPROVED** [review]

**Review by lunfardo314 — APPROVED** [review]

**Pull request merged**